0000-template.md

```diff
@@ -1,102 +1,118 @@
 
-# Rationale and Alternatives
+## Internal FAQ
 
-> - Why is this design the best in the space of possible designs?
-> - What other designs have been considered and what is the rationale for not
->   choosing them?
-> - What is the impact of not doing this?
+> The goal of this section is to help decide if this RFC should be implemented.
+> It should include answers to questions that the team is likely ask. Contrary
+> to the rest of the RFC, answers should be written "from the present" and
+> likely discuss design approach, implementation plans, alternative considered
+> and other considerations that will help decide if this RFC should be
+> implemented.
 
-# Adoption Strategy
+### Why are we doing this?
 
-> If we implement this proposal, how will existing CDK developers adopt it? Is
-> this a breaking change? How can we assist in adoption?
+> What is the motivation for this change?
 
-# Unresolved questions
+### Why should we _not_ do this?
 
-> - What parts of the design do you expect to resolve through the RFC process
->   before this gets merged?
-> - What parts of the design do you expect to resolve through the implementation
->   of this feature before stabilization?
-> - What related issues do you consider out of scope for this RFC that could be
->   addressed in the future independently of the solution that comes out of this
->   RFC?
+> Is there a way to address this use case with the current product? What are the
+> downsides of implementing this feature?
 
-# Future Possibilities
+### What changes are required to enable this change?
 
-> Think about what the natural extension and evolution of your proposal would be
-> and how it would affect CDK as whole. Try to use this section as a tool to more
-> fully consider all possible interactions with the project and ecosystem in your
-> proposal. Also consider how this fits into the roadmap for the project.
+> Briefly describe the high-level design approach for implementing this feature.
 >
-> This is a good place to "dump ideas", if they are out of scope for the RFC you
-> are writing but are otherwise related.
+> As appropriate, you can add an appendix with a more detailed design document.
 >
-> If you have tried and cannot think of any future possibilities, you may simply
-> state that you cannot think of anything.
+> This is a good place to reference a prototype or proof of concept, which is
+> highly recommended for most RFCs.
 
-# Implementation Plan
+### Is this a breaking change?
 
-INTENTIONALLY LEFT BLANK: an implementation plan will be added when the RFC is
-scheduled for implementation.
+> If the answer is no. Otherwise:
+>
+> Describe what ways did you consider to deliver this without breaking users?
+>
+> Make sure to include a `BREAKING CHANGE` section above with a a description of the breaking
+> changes and the migration path.
+
+### What alternative solutions did you consider?
+
+> Briefly describe alternative approaches that you considered.
```

Comment:
Brief description here with an optional appendix if it needs to be expanded.

Reply:
added